pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -124,36 +125,21 @@ type Interface interface {
 
 // New creates a new instance of DeployWaiter for the Etcd.
 func New(
-	c client.Client,
 	log logr.Logger,
+	c client.Client,
 	namespace string,
 	secretsManager secretsmanager.Interface,
-	role string,
-	class Class,
-	failureToleranceType *gardencorev1beta1.FailureToleranceType,
-	replicas *int32,
-	storageCapacity string,
-	defragmentationSchedule *string,
-	caRotationPhase gardencorev1beta1.ShootCredentialsRotationPhase,
-	k8sVersion string,
+	values Values,
 ) Interface {
-	name := "etcd-" + role
+	name := values.NamePrefix + "etcd-" + values.Role
```

Comment:
Instead of passing NamePrefix with `-` being appended at last, isn't it better to handle it here?

Reply:
Hm, why? This `etcd` component should probably not make any assumptions about the prefix. Maybe the caller doesn't want this `-` at the end of the prefix. I think it's more flexible to keep it like this.